### DISCUSSION TITLE
What's the routine to install a plugin in Honkit?

### DISCUSSION THREAD

**Discussion by zhzzy** [honkit/honkit#399] [Q&A] [answered]
I am using Honkit for the first time and want to install the Katex plugin in Honkit. I've read the plugin part of the document but I still met some problems. The following is what I've done.
``.
- I first installed the Honkit using `npm install honkit --save-dev`;
- Then I initialized a book using `npx honkit init`;
- Now, I want to use some plugins, for example, the Katex. I ran the command `npm i honkit-plugin-katex`;
- Then I add the following to my book.json file
```
{
    "plugins": ["katex"]
}
```
- Then I run `npx gitbook install` (Maybe this step is wrong?)

However, I've met the problem: `ReferenceError: Failed to load HonKit's plugin module: "katex" is not found.`

Then, I tried other plugins such as the `todo` plugin. I still get a similar error.  **So I want to know**, is this the correct method to install a plugin?

**Comment by azu** [comment]
> Then I run `npx gitbook install` (Maybe this step is wrong?)

Yes. This step is not required in HonKit.
(This step was required in Legacy's Gitbook, but it is not required in HonKitkit because it uses npm command.)

Can you make a reproduce repository?

I've tested `honkit-plugin-katex`, but it works.
https://stackblitz.com/edit/honkit-template-yb7hhp?file=book.json,README.md
![image](https://github.com/honkit/honkit/assets/19714/3dcaf38f-7b31-4ed6-89bf-4699acc1b800)

**Reply by zhzzy** [reply]
Thank you very much for your reply. I do the following two things:
1. I downloaded the honkit template you provided and it works well;
2. I made a reproduce repository, but the bug still exists. Here's the link:https://stackblitz.com/edit/honkit-template-iwpdtm?file=README.md

Again, I summarize all I've done in step 2.  
1. Starting from an empty file, I ran the command `honkit init`;
2. Then I create `book.json` file;
3. I added 
```json
{
    "plugins": ["katex"]
}
```
to my `book.json` file;
4. I ran the command `npm i honkit-plugin-katex`;
5. Then I ran the command `npm install`;
6. Finally, I ran `npx honkit serve`.

**Reply by azu** [reply, marked as answer]
Ah, You need to install `honkit` too.

```
# honkit
npm i honkit
# plugins
npm i honkit-plugin-katex
```

`npx honkit` do **download** and execute in temporarly dir.
(If `honkit` is installed in current `node_module/` dir, `npx honkit` just execute it.)


Probably, it will be instealled into different dirrectory between honkit it-self and plugins.
As a results, it cause the error `ReferenceError: Failed to load HonKit's plugin module: "katex" is not found.`

> ⚠️ Warning:
>
>    If you have installed honkit globally, you must install each plugins globally as well
>    If you have installed honkit locally, you must install each plugins locally as well
>
> We recommend installing honkit locally.
> https://github.com/honkit/honkit#installation


Instead of `npx`, You can use `npm exec honkit` in recent npm.
It does not download.

**Reply by zhzzy** [reply]
Thank you very much. It works.😊
